user/apps/l4test/kip.c

```diff
@@ -0,0 +1,292 @@
+/* Copyright (c) 2002, 2003, 2007, 2010 Karlsruhe University.
+ * All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#include <l4/ipc.h>
+#include <l4/kip.h>
+#include <l4/space.h>
+#include <l4/thread.h>
+#include <l4/arch.h>
+#include <l4/kdebug.h>
+#include <l4/pager.h>
+#include <l4io.h>
+#include <platform/cortex_m.h>
+
+#include "arch.h"
+#include "config.h"
+#include "l4test.h"
+#include "assert.h"
+
+__USER_TEXT
+int check_kipptr(L4_KernelInterfacePage_t *kip, void * ptr)
+{
+	if (((L4_Word_t)kip + (1 << kip->KipAreaInfo.X.s) >= (L4_Word_t)ptr) 
+		&& (ptr >= (void *) kip))
+		return 1;
+	printf("ERROR: kip-ptr (%p) not within KIP area\n", ptr);
+	return 0;
+}
+
+/* list of kernel ids, subids and what they are */
+typedef struct kid_list_t {
+	L4_Word_t id;
+	L4_Word_t subid;
+	const char *kernel;
+	const char * supplier;
+} kid_list_t;
+
+__USER_DATA
+kid_list_t kid_list[] = 
```

Comment:
We don't really want to implement full KernelInformation support. Use dummy array instead.